src/Components/NavbarComponent/ProfileDrawerComponent/ProfileDrawerComponent.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import {
+  Drawer,
+  List,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  type Theme,
+  useTheme,
+} from "@mui/material";
+import ManageAccountsIcon from "@mui/icons-material/ManageAccounts";
+import Brightness4Icon from "@mui/icons-material/Brightness4";
+import Brightness7Icon from "@mui/icons-material/Brightness7";
+import {COLORMODECONTEXT} from "../../../Scripts/ThemeContext";
+
+interface ProfileDrawerProps {
+  isOpen: boolean;
+  Close: (event: unknown, reason: "backdropClick" | "escapeKeyDown") => void;
+}
+
+export default function ProfileDrawerComponent(
+  props: Readonly<ProfileDrawerProps>
+): JSX.Element {
+  const {isOpen, Close} = props;
+
+  const SECTIONS: Array<{sectionName: string; sectionIcon: JSX.Element}> = [
```

Comment:
The same goes here, it might be a good idea to move the type definition of the object to a type file.